src/snowflake/snowpark/session.py

```diff
@@ -191,6 +194,29 @@ class Session:
     A :class:`Session` object is not thread-safe.
     """
 
+    class Decorator:
+        """A decorator class to translate connector messages to snowpark python client based exceptions """
+
+        @staticmethod
+        def wrap_exception(func):
+            def wrap(*args, **kwargs):
+                try:
+                    return func(*args, **kwargs)
+                except ProgrammingError as e:
+                    tb = sys.exc_info()[2]
+                    if "unexpected 'as'" in e.msg.lower():
+                        ne = (
+                            SnowparkClientExceptionMessages.SQL_PYTHON_REPORT_UNEXPECTED_ALIAS()
+                        )
+                        raise ne.with_traceback(tb) from None
```

Comment:
I see that `raise ... from None` suppresses the context, does https://github.com/snowflakedb/snowpark-python/blob/7672c690c9be1e75948aac1e7c1c55c89d7abf53/src/snowflake/snowpark/_internal/server_connection.py#L100 need this too then?

Reply:
I haven't touched the server_connection.py file yet. Should the exception wrapper here be refactored as well? It looks to me like separate logic from the rest, but we can add an `from None` to be consistent when it comes to context suppression. Thoughts?

Reply:
added context suppression to server_connection.py specific decorator.